Null Hypothesis ($H_0$): Device Type and Customer Satisfaction are independent.
Alternate Hypothesis ($H_1$): Device Type and Customer Satisfaction are dependent.

In [1]:
import numpy as np
import scipy.stats as stats
print("H0: Device Type and Customer Satisfaction are independent.")
print("H1: Device Type and Customer Satisfaction are dependent.\n")
observed_data = np.array([
    [50, 70],   
    [80, 100],  
    [60, 90],   
    [30, 50],   
    [20, 50]    
])
print("Observed Data Table:")
print(observed_data)

H0: Device Type and Customer Satisfaction are independent.
H1: Device Type and Customer Satisfaction are dependent.

Observed Data Table:
[[ 50  70]
 [ 80 100]
 [ 60  90]
 [ 30  50]
 [ 20  50]]


In [2]:
stat, p, dof, expected = stats.chi2_contingency(observed_data)
print("Chi-Square Statistic:", stat)
print("P-Value:", p)
print("Degrees of freedom:", dof)
print("Expected Table:\n", expected)

Chi-Square Statistic: 5.638227513227513
P-Value: 0.22784371130697179
Degrees of freedom: 4
Expected Table:
 [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


In [3]:
alpha = 0.05
critical = stats.chi2.ppf(1 - alpha, dof)
print("\nCritical Value:", critical)


Critical Value: 9.487729036781154


In [4]:
print("\nFinal Decision:")
if stat > critical:
    print("Reject H0: Device and Satisfaction are dependent.")
else:
    print("Fail to Reject H0: Device and Satisfaction are independent.")


Final Decision:
Fail to Reject H0: Device and Satisfaction are independent.


# Final Report: Chi-Square Test for Independence

### 1. Statistical Summary
* **Chi-Square Statistic:** 5.638
* **Critical Value (at 0.05 alpha):** 9.488
* **Degrees of Freedom:** 4
* **P-Value:** 0.227

### 2. Decision Analysis
We compared our Chi-Square statistic (5.638) against the critical value (9.488). Since our calculated statistic is **less than** the critical value, we **Fail to Reject the Null Hypothesis ($H_0$)**. 

### 3. Business Conclusion
There is no statistically significant association between the type of smart home device purchased (Smart Thermostat vs. Smart Light) and customer satisfaction. The data indicates that satisfaction levels are independent of the device type. Mizzare Corporation can conclude that both devices are performing consistently, and device type does not influence customer satisfaction.